#! /bin/sh /usr/share/dpatch/dpatch-run
## 15_fix_compile_warnings.dpatch by Erik Schanze <eriks@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixed warning, that NULL-pointer should not be 0 and
## DP: included header for exit()
## DP: Updated error string handling to strerror()

@DPATCH@
diff -urNad afio-2.5~/afio.c afio-2.5/afio.c
--- afio-2.5~/afio.c	2006-04-22 23:52:19.000000000 +0200
+++ afio-2.5/afio.c	2006-04-22 23:58:05.865771602 +0200
@@ -1589,7 +1589,7 @@
 	if(compressargs)
 	    execvp (compressprog, compress_arg_list);
 	else
-	    execlp (compressprog, compressprog, "-d", "-c", 0);
+	    execlp (compressprog, compressprog, "-d", "-c", NULL);
 	fprintf (stderr, "Could not uncompress, errno %d\n", errno);
 	exit(1);
 	break;
@@ -3014,7 +3014,7 @@
 	VOID dup (pfd[0]);
 	VOID close (pfd[0]);
 	
-        execlp (controlscript, controlscript, label, 0);
+        execlp (controlscript, controlscript, label, NULL);
 
 	warnarch("Problems running control script:",(off_t)0);	       
 	warn(controlscript,syserr());
@@ -3276,7 +3276,7 @@
 		  if(compressargs)
 		      execvp (compressprog, compress_arg_list);
 		  else
-		      execlp (compressprog, compressprog, "-d", "-c", 0);
+		      execlp (compressprog, compressprog, "-d", "-c", NULL);
 		  fprintf (stderr, "Could not uncompress, errno %d\n", errno);
 		  exit (1);
 	      }
@@ -4365,11 +4365,14 @@
 syserr ()
 {
   static char msg[40];
+  char *pTmp = NULL;
 
-  if (errno > 0 && errno < sys_nerr)
-    return ((char *) sys_errlist[errno]);
-  VOID sprintf (msg, "Unknown error (errno %d)", errno);
-  return (msg);
+  if (! (pTmp = strerror(errno)))
+  {
+    VOID sprintf (msg, "Unknown error (errno %d)", errno);
+    pTmp = msg;
+  }
+  return (pTmp);
 }
 
 /*
diff -urNad afio-2.5~/compfile.c afio-2.5/compfile.c
--- afio-2.5~/compfile.c	2006-04-22 23:52:18.000000000 +0200
+++ afio-2.5/compfile.c	2006-04-22 23:52:19.105067622 +0200
@@ -179,7 +179,7 @@
       if (open (name, O_RDONLY) >= 0)
       { 
 	  if(! compressargs)
-	      execlp (compressprog, compressprog, "-c", farg, 0);
+	      execlp (compressprog, compressprog, "-c", farg, NULL);
 	  else
 	      execvp (compressprog, compress_arg_list);
       }
diff -urNad afio-2.5~/regtest/cmpstat.c afio-2.5/regtest/cmpstat.c
--- afio-2.5~/regtest/cmpstat.c	2006-04-22 23:52:18.000000000 +0200
+++ afio-2.5/regtest/cmpstat.c	2006-04-22 23:52:19.105067622 +0200
@@ -11,6 +11,7 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <time.h>
+#include <stdlib.h>
 
 char ** av;
 
